### PR TITLE
[PGMORL] remaining iterations bugfix

### PR DESCRIPTION
**Pull request by mathieu-reymond**
I think I found a bug in the training iterations, since `self.iteration` starts at `self.warmup_iterations` after the warmup phase, it should be taken into account.

### CODE REVIEW

File: morl_baselines/multi_policy/pgmorl/pgmorl.py [hunk 1]
```diff
@@ -615,7 +615,7 @@ def train(self):
         # Evolution
         remaining_iterations = max(self.max_iterations - self.warmup_iterations, self.evolutionary_iterations)
```
Comment (ffelten) [inline morl_baselines/multi_policy/pgmorl/pgmorl.py:616]: @mathieu-reymond what do you think about fixing that line instead?

**Review by ffelten — COMMENTED** [review]
Thanks a lot 🫡

File: morl_baselines/multi_policy/pgmorl/pgmorl.py (hunk 1 above)
Comment (mathieu-reymond) [reply morl_baselines/multi_policy/pgmorl/pgmorl.py]: changed it, what do you think?

Comment (LucasAlegre) [reply morl_baselines/multi_policy/pgmorl/pgmorl.py]: It feels strange to me that the algorithm can run for more than `max_iterations` tho. Maybe we should rename those variables? Why do we need `max_iterations`? Can't the user provide only warmup and evolutionary iterations?

Comment (ffelten) [reply morl_baselines/multi_policy/pgmorl/pgmorl.py]: This is due to the way PGMORL sets the number of samples. On the one hand, it specifies a fixed number of timesteps. On the other hand, it asks to do `warmup` + `evolutionary` iterations of `steps_per_iteration`. The whole logic is very weird but that's how it is described in the paper. 
image

I scratched my head for some time to understand what they meant. And I saw the original implementation: https://github.com/mit-gfx/PGMORL/blob/master/morl/morl.py#L40

If you have suggestions to make the code easier I'm all for it.

**Comment by ffelten** [comment]
I checked out the code on my machine and the tests are passing. Very weird that the gh runner gets stuck...

File: morl_baselines/multi_policy/pgmorl/pgmorl.py (hunk 1 above)
Comment (LucasAlegre) [reply morl_baselines/multi_policy/pgmorl/pgmorl.py]: Ah, I thought max_iteration was a parameter of the algorithm, and not a internal variable. In that case, I think the current change is good

**Comment by mathieu-reymond** [comment]
@ffelten voila, hope everything is in order now

**Pull request merged**